src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -188,7 +188,7 @@ protected function addRoute(RouteCollection $collection, $annot, $globals, \Refl
                 continue;
             }
             foreach ($paths as $locale => $path) {
-                if (false !== strpos($path, sprintf('{%s}', $param->name))) {
+                if (preg_match(sprintf('/{%s(<[^>]+>)?}/', $param->name), $path)) {
```

Comment:
just in case, this match as in #26518:
`if (preg_match(sprintf('/{%s(?:<.*>)?}/', preg_quote($param->name)), $path)) {`